pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol

```diff
@@ -921,6 +937,111 @@ abstract contract ManagedPoolSettings is BasePool, ProtocolFeeCache, ReentrancyG
         }
     }
 
+    // Circuit Breakers
+
+    /**
+     * @notice Return the circuit breaker parameters for the given token.
+     * @dev These are the reference values (BPT price and weight complement) computed when the breaker was set,
+     * along with the percentage bounds. To get the current BPT price boundaries needed to check whether the
+     * circuit breaker should trip, call `getCurrentCircuitBreakerBounds`.
+     */
+    function getCircuitBreakerFields(IERC20 token)
+        external
+        view
+        returns (
+            uint256,
+            uint256,
+            uint256,
+            uint256
+        )
+    {
+        return CircuitBreakerLib.getCircuitBreakerFields(_circuitBreakerState[token]);
+    }
+
+    /**
+     * @notice Get the current BPT price bounds for the given token.
+     * @dev The bounds are set as percentages, but retrieved as BPT prices which can be directly compared to the
+     * current BPT price. Though the percentages are fixed when the breaker is set, the BPT price bounds are dynamic.
+     * If there is an ongoing gradual weight update, they will shift up or down with the current weight.
+     */
+    function getCurrentCircuitBreakerBounds(IERC20 token) public view returns (uint256, uint256) {
```

Comment:
```suggestion
    function getCurrentCircuitBreakerBounds(IERC20 token) external view returns (uint256, uint256) {
```
Lets have this external until (if) it's needed

Reply:
I am already using it in hasCircuitBreakerTripped